controller/controller_client.go

```diff
@@ -74,6 +74,7 @@ func (c *clientHandler) startShipper(unit *client.Unit) {
 		}
 		return
 	}
+	unit.RegisterDiagnosticHook("queue", "queue metrics", "queue_metrics.json", "application/json", c.outputHandler.QueueDiagCallback())
```

Comment:
We should add another integration test to `controller_client_test.go` to verify that the happy path for diags works e2e.